scripts/update_function_docs/update_function_docs.go

```diff
@@ -0,0 +1,294 @@
+// docs with the latest patch version for the release. If the docs are updated
+// then a commit is created with the changes. The manual steps left to the user
+// are to push the commit to a branch and create a pull request.
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"golang.org/x/mod/semver"
+	"gopkg.in/yaml.v2"
+)
+
+func exitWithErr(err error) {
+	fmt.Fprintf(os.Stderr, "%v\n", err)
+	os.Exit(1)
+}
+
+func runCmd(name string, arg ...string) (string, error) {
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	cmd := exec.Command(name, arg...)
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	fmt.Printf("%s\n", cmd.String())
+	err := cmd.Run()
+	if err != nil {
+		return stdout.String(), fmt.Errorf("%s\n%s", stderr.String(), err)
+	}
+	return stdout.String(), err
+}
+
+func isCleanRepo() bool {
+	_, err := runCmd("git", "diff-index", "--quiet", "HEAD", "--")
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+func gitFetch() error {
+	_, err := runCmd("git", "fetch", "--tags")
+	return err
+}
+
+func gitCheckout(branch string) error {
+	_, err := runCmd("git", "checkout", branch)
+	return err
+}
+
+func gitAdd() error {
+	_, err := runCmd("git", "add", "-u")
+	return err
+}
+
+func gitCommit(msg string) error {
+	formattedMsg := fmt.Sprintf("\"%s\"", msg)
+	stdout, err := runCmd("git", "commit", "-m", formattedMsg)
+	fmt.Printf("%v\n", stdout)
+	return err
+}
+
+func gitShow() error {
+	stdout, err := runCmd("git", "show")
+	fmt.Printf("%v\n", stdout)
+	return err
+}
+
+var (
+	releaseBranchPattern = regexp.MustCompile(`[-\w]*\/(v\d*\.\d*)`)
+	releaseTagPattern    = regexp.MustCompile(`.*(go|ts)\/[-\w]*\/(v\d*\.\d*\.\d*)`)
+)
+
+func dirExists(path string) bool {
+	if stat, err := os.Stat(path); err == nil && stat.IsDir() {
+		return true
+	}
+	return false
+}
+
+type functionRelease struct {
+	FunctionName       string
+	MinorVersion       string
+	Language           string
+	LatestPatchVersion string
+	FunctionPath       string
+	ExamplePaths       []string
+}
+
+// newFunctionRelease allocates and initializes a functionRelease
+func newFunctionRelease(branch string) (*functionRelease, error) {
+	fr := &functionRelease{}
+	if !releaseBranchPattern.MatchString(branch) {
+		return nil, fmt.Errorf("invalid branch format")
+	}
+	segments := strings.Split(branch, "/")
+	fr.MinorVersion = segments[len(segments)-1]
+	fr.FunctionName = segments[len(segments)-2]
+	if err := fr.readLatestPatchVersion(); err != nil {
+		return nil, err
+	}
+	if err := fr.readDocPaths(); err != nil {
+		return nil, err
+	}
+	return fr, nil
+}
+
+// readLatestPatchVersion of the release from git tags
+func (fr *functionRelease) readLatestPatchVersion() error {
+	if fr.FunctionName == "" || fr.MinorVersion == "" {
+		return fmt.Errorf("missing function name and/or minor version")
+	}
+	tags, err := runCmd("git", "tag")
```

Comment:
You have helper fn for all other git commands. Why not add one?

Reply:
Done